man/fi_av.3.md

```diff
@@ -449,6 +466,36 @@ soon as they have been added to an AV, even if asynchronous operation
 has been specified.  Similarly, a provider may lazily release
 resources from removed entries.
 
+# USER IDENTIFIERS FOR ADDRESSES
+
+As described above, endpoint addresses that are inserted into an AV are
+mapped to an fi_addr_t value.  The fi_addr_t is used in data transfer APIs
+to specify the destination of an outbound transfer, in receive APIs to
+indicate the source for an inbound transfer, and also in completion events
+to report the source address of inbound transfers.  The FI_AV_USER_ID
+capability bit and flag provide a mechanism by which the fi_addr_t value
+reported by a completion event is replaced with a user-specified value
+instead.  This is useful for applications that need to map the source
+address to their own data structure.
+
+Support for FI_AV_USER_ID is provider specific, as it may not be feasible
+for a provider to implement this support without significant overhead.
+For example, some providers may need to add a reverse lookup mechanism.
+This feature may be unavailable if shared AVs are requested, or
+negatively impact the per process memory footprint if implemented.  For
+providers that do not support FI_AV_USER_ID, users may be able to trade
+off lookup processing with protocol overhead, by carrying source
+identification within a message header.
+
+User-specified fi_addr_t values are provided as part of address insertion
+(e.g. fi_av_insert).  When the FI_AV_USER_ID flag is passed to any of the
+insert calls, the caller must specify an fi_addr_t identifier value to
+associate with each address.  The provider will record that identifier and
```

Comment:
Is it through the `fi_addr` parameter of the `fi_av_insert*` calls? A note under the description of that parameter (L89) would be helpful.

Reply:
Yes.  I was speaking with @aingerson about this, and at the time defined a new call  (fi_av_set_context) to set this.  But after looking at this more, that would just result in the user needing to make a second call after calling insert.